src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -469,8 +469,46 @@ public bool TryGetDateTime(out DateTime value)
                 throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
             }
 
-            ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
-            return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
+            ReadOnlySpan<byte> span = stackalloc byte[0];
+
+            if (HasValueSequence)
+            {
+                int sequenceLength = (int)ValueSequence.Length;
+
+                if (!JsonReaderHelper.IsValidDateTimeParsingLength(sequenceLength))
+                {
+                    value = default;
+                    return false;
+                }
+
+                Debug.Assert(sequenceLength <= JsonConstants.MaximumEscapedDateTimeOffsetParseLength);
+                Span<byte> stackSpan = stackalloc byte[sequenceLength];
```

Comment:
Is it possible to re-use the outer span and remove `stackSpan all together?
```C#
span = stackalloc byte[sequenceLength];
ValueSequence.CopyTo(stackSpan);
```

Reply:
No. We can't copy to `ReadOnlySpan` span. Also can't change span to `Span` because we wouldn't be able to assign `ReadOnlySpan` ValueSpan to it.